networking/configuring-a-custom-pki.adoc

```diff
@@ -0,0 +1,57 @@
+[id="configuring-a-custom-pki"]
+= Configuring a custom PKI
+include::modules/common-attributes.adoc[]
+:context: configuring-a-custom-pki
+
+toc::[]
+
+Some platform components (i.e., web console) use Routes for communication and,
+therefore, must trust other components' certificates to interact with them. If
+you're using a custom public key infrastructure (PKI), you must configure it so
+its privately signed CA certificates are recognized across the cluster.
+
+You can leverage the Proxy API to add cluster-wide trusted CA certificates. You
+must do this either during installation *or* at runtime.
+
+* During *installation*, configure the cluster-wide proxy. You must define your
+privately signed CA certificates in the `install-config.yaml` file's
+`additionalTrustBundle` setting.
++
+The installation program generates a ConfigMap that is named `user-ca-bundle`
+that contains the additional CA certificates you defined. The Cluster Network
+Operator then creates a `trusted-ca-bundle` ConfigMap that merges these CA
+certificates with the {op-system-first} trust bundle; this ConfigMap is
+referenced in the Proxy object's `trustedCA` field.
+
+* At *runtime*, modify the default Proxy object to include your privately signed
+CA certificates (part of cluster's proxy enablement workflow). This involves
+creating a ConfigMap that contains the privately signed CA certificates that
+should be trusted by the cluster, and then modifying the proxy resource with the
+`trustedCA` referencing the privately signed certificates' ConfigMap.
+
+[NOTE]
+====
+The installer configuration's `additionalTrustBundle` field and the proxy
+resource's `trustedCA` field are used to manage the cluster-wide trust bundle;
+`additionalTrustBundle` is used at install time and the proxy's `trustedCA` is
+used at runtime.
+
+The `trustedCA` field is a reference to a `ConfigMap` containing the custom
+certificate and key pair used by the cluster component.
+====
+
+include::modules/installation-configure-proxy.adoc[leveloffset=+1]
+
+include::modules/nw-proxy-configure-object.adoc[leveloffset=+1]
+
+== Checkpoint
```

Comment:
I'd argue that this Checkpoint section isn't really adding any value, and I don't think we normally do something like this.

I have seen places that have something like "Verifying <x>" that verifies something you did earlier, but it usually has actual steps to check something.

But for your situation, I'd think this just wasn't necessary and that the lead-in from the assembly should be sufficient, if you link to these.

Reply:
@bergerhoffer makes sense. My intention was to call out the specific places where we were adding private certs, since some parts of the included modules, while necessary, were more about the proxy set up and less about the cert specification.

I agree it's a bit out of place compared to what I've seen in other docs. I'll remove if you think the current setup is clear enough. :+1:

Reply:
Just so that I'm clear, the procedure modules you included for install/configure the proxy.. the user still needs to follow the entire procedures, correct?

Wanted to make sure that they don't only need to follow just a piece of the procedures (because then I'd suggest making your own steps instead of reusing these).

Reply:
Yes, they need the entire procedure modules, which go through setting up the proxy. So although the cert specifications are the highlighted part of the process, the proxy must still be configured also.